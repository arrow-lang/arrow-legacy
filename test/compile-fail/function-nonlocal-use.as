let a: int;

def do(_: int) { }

// use: `a`
def methodA() {
  if false {
    methodC();
  } else {
    methodB();
  }
}

methodB(); // error

// use: `a`
def methodC() {
  methodA();
  do(a);
}

// use: `a`
def methodB() {
  methodC();
}
