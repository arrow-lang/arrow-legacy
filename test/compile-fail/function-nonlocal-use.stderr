[0;37mtest/compile-fail/function-nonlocal-use.as:14.1-10: [0;31merror: [1;37muse of possibly uninitialized variable 'a'[0m
